plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -121,6 +126,11 @@
 
     private static final Joiner DOT_JOINER = Joiner.on(".");
 
+    private final Cache<String, Boolean> SNAPSHOT_ISOLATION_ENABLED = CacheBuilder.newBuilder()
+            .maximumSize(1)
+            .expireAfterWrite(ofSeconds(30))
```

Comment:
Make the cache key a singleton, otherwise the max size looks fishy.

```
private enum SnapshotIsolationEnabledCacheKey
{
 // The snapshot isolation can be enabled or disabled on database level. We connect to single
 // database, so from our perspective, this is a global property.
 INSTANCE
}
```

Cache for 5 minutes?
i bet this is very very slowly changing state.